extensions/shared/upgrade-nudge/index.jsx

```diff
@@ -0,0 +1,62 @@
+/**
+ * External dependencies
+ */
+import { get, startsWith } from 'lodash';
+import { __, sprintf } from '@wordpress/i18n';
+import { addQueryArgs } from '@wordpress/url';
+import { Button } from '@wordpress/components';
+import { withSelect } from '@wordpress/data';
+import Gridicon from 'gridicons';
+
+/**
+ * Internal dependencies
+ */
+import getSiteFragment from '../get-site-fragment';
+import './store';
+
+import './style.scss';
+
+const UpgradeNudge = ( { planName, planPathSlug, postId, postType } ) => (
```

Comment:
I see quite a long delay between "Upgrade" button pressed and the actual redirection. Is this something that we should resolve in this PR? 

Also, this seems kinda expected, that upgrade flow will be loaded in the same tab, but in case of unsaved changes/updates - leaving the editor page is quite scary. Maybe we can trigger autosave or override default alert, to make sure folks are not confused? 

Video for reference: 
![screencast 2019-07-04 17-18-10](https://user-images.githubusercontent.com/5654161/60673555-abfd0880-9e80-11e9-8390-bf4dd3c67945.gif)

Reply:
I agree it's scary. We'll see how we can ease it, perhaps by saving a revision first.

Just saving edits made in already published posts would be problematic.